packages/headless/src/controllers/relevance-inspector/headless-relevance-inspector.ts

```diff
@@ -148,19 +148,30 @@ export function buildRelevanceInspector(
     },
 
     subscribe(listener: () => void) {
-      listener();
-      return engine.subscribe(() => {
-        if (hasNewResponse(engine.state.search.response.searchUid)) {
-          if (
-            this.state.isEnabled &&
-            (options.automaticallyLogInformation || logOnNextResponse)
-          ) {
-            logOnNextResponse = false;
-            this.logInformation();
-          }
-          listener();
+      const unsubscribeLogListener = engine.subscribe(() => {
+        if (
+          hasNewResponse(engine.state.search.response.searchUid) &&
+          this.state.isEnabled &&
+          (options.automaticallyLogInformation || logOnNextResponse)
+        ) {
+          logOnNextResponse = false;
+          this.logInformation();
         }
       });
+
+      const getState = () => this.state;
+
+      const unsubscribeStateListener = {
+        ...controller,
+        get state() {
+          return getState();
```

Comment:
Could we just return `this.state` in here and remove the `getState` function?

Reply:
Inside the getter, `this.state` seems to be redefined.

Reply:
ok, I guess this is the best we can do then 👍